sections/cart-drawer.liquid

```diff
@@ -0,0 +1,334 @@
+<script src="{{ 'cart.js' | asset_url }}" defer="defer"></script>
+
+<style>
+  .drawer {
+    visibility: hidden;
+  }
+</style>
+
+<cart-drawer class="drawer{% if cart == empty %} is-empty{% endif %}">
```

Comment:
Did we consider at any point using the `details` / `summary` pattern on this component, similarly to what we're doing in menu drawers and search? The a11y properties could then be applied via JS to make sure it conveys the correct semantics.

One benefit of that would be for slow connections. Currently, if you tap the cart icon before the scripts finish loading it'll redirect you to the `cart` page. Redirecting to another page isn't really something good for users on slow connections.

Without knowing the full details behind this decision to not use this pattern, I think refactoring this as `details`/`summary` could result in a better experience while keeping our web native + progressive enhancement principles.

This is more a suggestion for improvement in the form of a question than a must have to ship, btw.

Reply:
Oh, agree 100% with your comment, Scott. Thanks for this perfect explanation. 

Looking back at my comment, I didn't really realize that displaying a `details`/`summary` would basically result in displaying a cart drawer that is not really functional at all without JS. That alone makes redirecting to the cart page a better choice. It's pointless to show the user a certain component that does not provide any functionality and is also using some pretty confusing semantics. This is also a key difference between this component and the search menu or the navigation links, which "works" without JS - they're just enhanced by JS.

The better A11Y UX is where I thought the JS could be used as a progressive enhancement, but I now see that we already have a dedicated page that already solves this problem haha. 

I'm resolving this conversation, please my first comment in this thread above.